### DISCUSSION TITLE
RadioMaster T8 Lite as gamepad for cfclient

### DISCUSSION THREAD

**Discussion by JBuzzatto** [bitcraze/discussions#886] [General]
Hi,

I have a RadioMaster T8 Lite radio controller, and I would like to use it as the gamepad input to fly a crazyflie 2.1 in assist mode. 

My setup includes running cfclient.gui on my Windows 10 computer (no virtual machine), the crazyradio 2.0, and a crazyflie 2.1 with the Flow Deck V2. 

Currently, the client can recognize the sticks from the radio, but not the switches it has. Is there an easy way to change the cfclient.gui script to interpret one of the switches as a button to change to assist mode?

Thank you for the help.

**Comment by tobbeanton** [comment]
> Currently, the client can recognize the sticks from the radio, but not the switches it has. Is there an easy way to change the cfclient.gui script to interpret one of the switches as a button to change to assist mode?

The gamepad code is a bit complicated so might be easier to attack the lower layer. Perhaps there is a way in windows to make the switches behave like buttons?

**Reply by JBuzzatto** [reply]
Hi Tobias,

Thank you for getting back to me. I tried making the switches behave like buttons on Windows but had no success. 

However, I did solve my problem by changing the client code. 
I just added the following line in the input readers code for pysdl2 (https://github.com/bitcraze/crazyflie-clients-python/blob/master/src/cfclient/utils/input/inputreaders/pysdl2.py) 

```
def read(self):
        while not self._event_queue.empty():
            e = self._event_queue.get_nowait()
            if e.type == sdl2.SDL_JOYAXISMOTION:
                self.axes[e.jaxis.axis] = e.jaxis.value / 32767.0

            if e.type == sdl2.SDL_JOYBUTTONDOWN:
                self.buttons[e.jbutton.button] = 1

            if e.type == sdl2.SDL_JOYBUTTONUP:
                self.buttons[e.jbutton.button] = 0

            if e.type == sdl2.SDL_JOYHATMOTION:
                self._set_virtual_dpad(e.jhat.value)
            
            self.buttons[0] = self.axes[5] # <- added line
        return [self.axes, self.buttons]
```

That made the switch corresponding to axis 5 to behave as a button on the client. It worked for setting up and switching to assisted mode.
